core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -93,6 +93,8 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
 
   val retainedStages = conf.getInt("spark.ui.retainedStages", SparkUI.DEFAULT_RETAINED_STAGES)
   val retainedJobs = conf.getInt("spark.ui.retainedJobs", SparkUI.DEFAULT_RETAINED_JOBS)
+  val retainedTasks = conf.getInt("spark.ui.retainedTasks", SparkUI.DEFAULT_RETAINED_TASKS)
```

Comment:
`retainedTasks` is compared to `stageData.taskData.size`, which is a `int`, I don't see how `TaskId` being a long is relevant. 

And I've switched to a `ConfigEntry` and will push my changes after testing.